src/modules/Crowbar.rb

```diff
@@ -266,22 +263,35 @@ def Write
         Report.Error(Message.ErrorWritingFile(@crowbar_file))
       end
 
+      if FileUtils.Exists(@repos_file)
+        default_repos = yaml2hash(@repos_file)
+      else
+        default_repos = {}
+      end
+
       @repos.each do |product_name, product|
+        default_repos_prod = default_repos[product_name] || {}
         product.each do |repo_name, repo|
-          # remove empty repo definitions
-          if repo["url"].empty? && !(repo["ask_on_error"] || false)
-            @repos[product_name].delete(repo_name)
-          # remove just url if it is empty and non-default ask_on_error stays
-          elsif repo["url"].empty?
-            @repos[product_name][repo_name].delete("url")
+          if default_repos_prod.key? repo_name
+            # remove repos that have no change compared to defaults
+            unless (default_repos_prod[repo_name]["url"] || "") != (repo["url"] || "") ||
+                (default_repos_prod[repo_name]["ask_on_error"] || false) != (repo["ask_on_error"] || false)
```

Comment:
this condition looks a bit complex for one line, would it make sense to put it into a function with a proper name?